Spigot-Server-Patches/0581-Improve-ServerGUI.patch

```diff
@@ -0,0 +1,121 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: AlexProgrammerDE <40795980+AlexProgrammerDE@users.noreply.github.com>
+Date: Sat, 3 Oct 2020 08:27:40 +0200
+Subject: [PATCH] Improve ServerGUI
+
+- Added logo to server frame
+- Show tps in the server stats
+
+diff --git a/src/main/java/net/minecraft/server/GuiStatsComponent.java b/src/main/java/net/minecraft/server/GuiStatsComponent.java
+index 859e31c63f94bdc7729c6d475990750b76e24b9c..3c98a6feded87f93c0119e2d414779037f003c66 100644
+--- a/src/main/java/net/minecraft/server/GuiStatsComponent.java
++++ b/src/main/java/net/minecraft/server/GuiStatsComponent.java
+@@ -1,5 +1,7 @@
+ package net.minecraft.server;
+ 
++import org.bukkit.Bukkit;
++
+ import java.awt.Color;
+ import java.awt.Dimension;
+ import java.awt.Graphics;
+@@ -16,7 +18,7 @@ public class GuiStatsComponent extends JComponent {
+     });
+     private final int[] b = new int[256];
+     private int c;
+-    private final String[] d = new String[11];
++    private final String[] d = new String[12];
+     private final MinecraftServer e;
+     private final Timer f;
+ 
+@@ -35,8 +37,13 @@ public class GuiStatsComponent extends JComponent {
+     private void b() {
+         long i = Runtime.getRuntime().totalMemory() - Runtime.getRuntime().freeMemory();
+ 
++        // Paper start - Add tps entry
++        double tps = Bukkit.getTPS()[0];
+         this.d[0] = "Memory use: " + i / 1024L / 1024L + " mb (" + Runtime.getRuntime().freeMemory() * 100L / Runtime.getRuntime().maxMemory() + "% free)";
+         this.d[1] = "Avg tick: " + GuiStatsComponent.a.format(this.a(this.e.h) * 1.0E-6D) + " ms";
++        this.d[2] = "TPS:" + ( ( tps > 21.0 ) ? "*" : "" ) + Math.min( Math.round( tps * 100.0 ) / 100.0, 20.0 );
```

Comment:
The TPS should be the same as with `/tps`; the last 5s isn't always the best metric.

Also use an obfhelper to access `this.d`.

Reply:
1. So you want Bukkit.getServer().getTPS[0] instead of Bukkit.getTPS[0] ?
2. What is a obfhelper? I just searched it up on google, but didn't anything useful.

Reply:
1. Check the TPS command's source.
1. Check the CONTRIBUTING.md file.

Reply:
@Proximyst 
1. You mean this code?
```java
double[] tps = org.bukkit.Bukkit.getTPS();
        String[] tpsAvg = new String[tps.length];

        for ( int i = 0; i < tps.length; i++) {
            tpsAvg[i] = format( tps[i] );
        }
        sender.sendMessage(ChatColor.GOLD + "TPS from last 1m, 5m, 15m: " + org.apache.commons.lang.StringUtils.join(tpsAvg, ", "));```
I still don't get what youwant me to do here. 
2. Like this? `public String[] getStatEntries() { return this.d; } // Paper - OBFHELPER`